<!DOCTYPE html>
<html class="writer-html5" lang="en" >
<head>
  <meta charset="utf-8" /><meta name="generator" content="Docutils 0.18.1: http://docutils.sourceforge.net/" />

  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  <title>DRExM³L Drug REpurposing using Mechanistic Models of signal transduction and eXplainable Machine Learning &mdash; drexml 1.1.0+editable documentation</title>
      <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
      <link rel="stylesheet" href="_static/css/theme.css" type="text/css" />
  <!--[if lt IE 9]>
    <script src="_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
        <script data-url_root="./" id="documentation_options" src="_static/documentation_options.js"></script>
        <script src="_static/jquery.js"></script>
        <script src="_static/underscore.js"></script>
        <script src="_static/_sphinx_javascript_frameworks_compat.js"></script>
        <script src="_static/doctools.js"></script>
        <script src="_static/sphinx_highlight.js"></script>
    <script src="_static/js/theme.js"></script>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="CLI Reference" href="cli-reference.html" /> 
</head>

<body class="wy-body-for-nav"> 
  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >

          
          
          <a href="#" class="icon icon-home">
            drexml
          </a>
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" aria-label="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>
        </div><div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <ul>
<li class="toctree-l1"><a class="reference internal" href="cli-reference.html">CLI Reference</a></li>
</ul>
<ul>
<li class="toctree-l1"><a class="reference internal" href="api_reference.html">API Reference</a></li>
</ul>
<ul>
<li class="toctree-l1"><a class="reference internal" href="cli-reference.html">CLI Reference</a></li>
</ul>
<ul>
<li class="toctree-l1"><a class="reference internal" href="authors-reference.html">Authors and contributors</a></li>
</ul>

        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap"><nav class="wy-nav-top" aria-label="Mobile navigation menu" >
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="#">drexml</a>
      </nav>

      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="Page navigation">
  <ul class="wy-breadcrumbs">
      <li><a href="#" class="icon icon-home" aria-label="Home"></a></li>
      <li class="breadcrumb-item active">DRExM³L Drug REpurposing using Mechanistic Models of signal transduction and eXplainable Machine Learning</li>
      <li class="wy-breadcrumbs-aside">
            <a href="_sources/index.rst.txt" rel="nofollow"> View page source</a>
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
             
  <section id="drexm3l-drug-repurposing-using-mechanistic-models-of-signal-transduction-and-explainable-machine-learning">
<h1>DRExM³L Drug REpurposing using Mechanistic Models of signal transduction and eXplainable Machine Learning<a class="headerlink" href="#drexm3l-drug-repurposing-using-mechanistic-models-of-signal-transduction-and-explainable-machine-learning" title="Permalink to this heading"></a></h1>
<p>DRExM³L or drexml is a Python package for finding links between Known Drug Targets (KDTs)
and signaling circuits.</p>
<figure class="align-center" id="id1">
<img alt="_images/holird.png" src="_images/holird.png" />
<figcaption>
<p><span class="caption-text">Holistic Drug REpurposing Model</span><a class="headerlink" href="#id1" title="Permalink to this image"></a></p>
</figcaption>
</figure>
</section>
<section id="installation">
<h1>Installation<a class="headerlink" href="#installation" title="Permalink to this heading"></a></h1>
<p>You can install drexml using:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">pip</span> <span class="n">install</span> <span class="n">drexml</span>
</pre></div>
</div>
</section>
<section id="getting-started">
<h1>Getting started<a class="headerlink" href="#getting-started" title="Permalink to this heading"></a></h1>
<p>DRExML is a Python software package that allows you to explore and contextualize the pharmacological landscape of a disease characterized by its signaling circuits.</p>
<p>Mechanistic modeling allows us to analyze the activity of signaling circuits that constitute biological pathways from gene expression data. This package uses machine learning models to infer the potential regulatory properties of known drug targets (KDTs) in these signaling circuits. This exploration of drug landscape from the disease map allows the identification of significantly influential KDT-targeting drugs that could play a therapeutic role in that disease.</p>
<p>The user can upload information about their disease to build their mechanistic map (see Input). The package contains information on the gene expression levels normalized by the edgeR package of the KDTs (from GTEx Analysis Release V8; dbGaP 110 Accession phs000424.v8.p2) ) and the drugs that interact with each KDT (from DrugBank version 05.01.10). Finally, default signal transduction values are computed using the Hipathia package (version 2.14.0).</p>
<p>For more information on the command line interface, check the <a class="reference internal" href="cli-reference.html#cli-reference"><span class="std std-ref">CLI Reference</span></a>.</p>
<div class="toctree-wrapper compound">
</div>
<p>Default transcriptomic profiles are obtained from the GTEx portal (GTEx Analysis Release V8; dbGaP 110 Accession phs000424.v8.p2) and normalized using the edgeR package (version 3.40.0). Default KDTs are obtained from the DrugBank database (version 05.01.10). Finally, default signal transduction values are computed using the Hipathia package (version 2.14.0).</p>
<p>For more information, check the <a class="reference internal" href="api_reference.html#api-reference"><span class="std std-ref">API Reference</span></a>.</p>
<div class="toctree-wrapper compound">
</div>
</section>
<section id="highlights">
<h1>Highlights<a class="headerlink" href="#highlights" title="Permalink to this heading"></a></h1>
<ul class="simple">
<li><p>Contextualize the pharmacological landscape of a mechanistically characterized disease.</p></li>
<li><p>Obtain a list of KDTs which show a considerable influence over a given disease map</p></li>
<li><p>Quick visualization of the knowledge: generation of figures and tables of results</p></li>
</ul>
</section>
<section id="input">
<h1>Input<a class="headerlink" href="#input" title="Permalink to this heading"></a></h1>
<p>The disease definition file is the main point of entry to interact with the software. Here, variables are defined pertaining to the disease map and activity matrix, as well as to the KDTs explored.</p>
<p>Main variables pertaining to the construction of the disease map and activity matrix are:</p>
<ul class="simple">
<li><p>seed_genes: list of Entrez identifiers of genes suspected to be involved in the disease. The algorithm returns KEGG subpathways which contain at least on of these genes as the disease map</p></li>
<li><p>disease_id: UMLS CUI disease identifiers. The algorithm returns genes associated to this disease using the curated DISGENET database and returns the KEGG subpathways which contain at least one of these genes as the disease map</p></li>
<li><p>circuits: .tsv file with a list of circuits composing the target disease map, identified by their KEGG ID.</p></li>
</ul>
<p>The path to each of these three files must be defined in the environment file (e.g. disease.env). Note that users can use different signal transduction algorithms and/or different KDT transcriptomic profiles to explore. These must be given as TSV files by providing a path to the activity matrix through the “pathvals” variable and to the KDTs through the “gene_exp” variable.</p>
</section>
<section id="output">
<h1>Output<a class="headerlink" href="#output" title="Permalink to this heading"></a></h1>
<p>Software outputs are stored as TSV files:</p>
<ul class="simple">
<li><p>shap_summary_symbol.tsv: final relevance matrix (SHAP values attributed to KDTs)</p></li>
<li><p>shap_selection_symbol.tsv: binary matrix indicating the KDTs have been selected for a given circuit or not</p></li>
<li><p>stability_results_symbol.tsv: R2 and stability estimates of circuits along with their confidence intervals</p></li>
</ul>
</section>
<section id="visualization">
<h1>Visualization<a class="headerlink" href="#visualization" title="Permalink to this heading"></a></h1>
<p>The plot sub-command can be used to generate a number of summary graphics. Funcionalities include “plot_metrics” through which R2 and stability estimates can be visualized and “plot_relevance_heatmap” through which a heat map of the KDT SHAP scores can be generated in order to visualize KDT relevance over circuits.</p>
<p>For more information on the command line interface, check the <a class="reference internal" href="cli-reference.html#cli-reference"><span class="std std-ref">CLI Reference</span></a>.</p>
<div class="toctree-wrapper compound">
</div>
<p>For more information on the authors, check the <a class="reference internal" href="authors-reference.html#authors-reference"><span class="std std-ref">Authors and contributors</span></a>.</p>
<div class="toctree-wrapper compound">
</div>
</section>
<section id="indices-and-tables">
<h1>Indices and tables<a class="headerlink" href="#indices-and-tables" title="Permalink to this heading"></a></h1>
<ul class="simple">
<li><p><a class="reference internal" href="genindex.html"><span class="std std-ref">Index</span></a></p></li>
<li><p><a class="reference internal" href="py-modindex.html"><span class="std std-ref">Module Index</span></a></p></li>
<li><p><a class="reference internal" href="search.html"><span class="std std-ref">Search Page</span></a></p></li>
</ul>
</section>


           </div>
          </div>
          <footer><div class="rst-footer-buttons" role="navigation" aria-label="Footer">
        <a href="cli-reference.html" class="btn btn-neutral float-right" title="CLI Reference" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right" aria-hidden="true"></span></a>
    </div>

  <hr/>

  <div role="contentinfo">
    <p>&#169; Copyright 2022, Carlos Loucera.</p>
  </div>

  Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    provided by <a href="https://readthedocs.org">Read the Docs</a>.
   

</footer>
        </div>
      </div>
    </section>
  </div>
  <script>
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script> 

</body>
</html>